docs/usage/deriving-data-selectors.md

```diff
@@ -0,0 +1,638 @@
+
+This means that if you want to pass additional parameters through to the output selector, you must define input selectors that extract those values from the original selector arguments:
+
+```js
+const selectItemsByCategory = createSelector(
+  [
+    // Usual first input - extract value from `state`
+    state => state.items,
+    // Take the second arg, `category`, and forward to the output selector
+    (state, category) => category
+  ],
+  // Output selector gets (`items, category)` as args
+  (items, category) => items.filter(item => item.category === category)
+)
+```
+
+For consistency, you may want to consider passing additional parameters to a selector as a single object, such as `selectThings(state, otherArgs)`, and then extracting values from the `otherArgs` object.
+
+#### Selector Factories
+
+**`createSelector` only has a default cache size of 1, and this is per each unique instance of a selector**. This creates problems when a single selector function needs to get reused in multiple places with differing inputs.
+
+One option is to create a "selector factory" - a function that runs `createSelector()` and generates a new unique selector instance every time it's called:
+
+```js
+const makeSelectItemsByCategory = () => {
+  const selectItemsByCategory = createSelector(
+    [state => state.items, (state, category) => category],
+    (items, category) => items.filter(item => item.category === category)
+  )
+  return selectItemsByCategory
+}
+```
+
+This is particularly useful when multiple similar UI components need to derive different subsets of the data based on props.
+
+## Alternative Selector Libraries
+
+While Reselect is the most widely used selector library with Redux, there are many other libraries that solve similar problems, or expand on Reselect's capabilities.
+
+### `proxy-memoize`
+
+`proxy-memoize` is a relatively new memoized selector library that uses a unique implementation approach. It relies on ES6 `Proxy` objects to track attempted reads of nested values, then compares only the nested values on later calls to see if they've changed. This can provide better results than Reselect in some cases.
+
+A good example of this is a selector that derives an array of todo descriptions:
+
+```js
+import { createSelector } from 'reselect'
+
+const selectTodoDescriptionsReselect = createSelector(
+  [state => state.todos],
+  todos => todos.map(todo => todo.text)
+)
+```
+
+This will unfortunately recalculate the derived array if any other value inside of `state.todos` changes, such as toggling a `todo.completed` flag. The _contents_ of the derived array are identical, but because the input `todos` array changed, it has to calculate a new output array, and that has a new reference.
```

Comment:
```suggestion
Unfortunately, this will recalculate the derived array if any other value inside of `state.todos` changes, such as toggling a `todo.completed` flag. The _contents_ of the derived array are identical, but because the input `todos` array changed, it has to calculate a new output array, and that has a new reference.
```